test/logflare_web/controllers/log_controller_test.exs

```diff
@@ -293,4 +296,49 @@ defmodule LogflareWeb.LogControllerTest do
     reject(&Users.preload_billing_account/1)
     :ok
   end
+
+  describe "benchmarks" do
+    setup [:v1_pipeline_setup, :warm_caches, :reject_context_functions]
+
+    setup %{user: user, conn: conn} do
+      conn = put_req_header(conn, "x-api-key", user.api_key)
+
+
+      :timer.sleep(1500)
+      {:ok, user: user, conn: conn}
+    end
+
+    @tag :benchmark
+    test "v1 ingestion", %{conn: conn, source: source} do
```

Comment:
Does it make sense to make it part of test suite? Wouldn't it be better to have it as a set of separate files defined somewhere?

Reply:
it is for reusing the setups for mocks and fixtures, which lets us benchmark the larger whole of the pipeline, as opposed to small parts.

smaller functions that are isolated to a module for example can be put in a separate script.

Reply:
Hmm, ok. But I would think about [registering custom test type](https://hexdocs.pm/ex_unit/ExUnit.Case.html#register_test/4) for that case. This probably would make it a little bit easier to spot that it is not real test.

Reply:
yea we can do that once we have a larger collection of benchmarks suites.